middleware/2fa.js

```diff
@@ -0,0 +1,303 @@
+const nearAPI = require('near-api-js');
+const { utils: { serialize: { base_encode } } } = nearAPI;
+const crypto = require('crypto');
+const nacl = require('tweetnacl');
+const { creatorKeyJson } = require('./near');
+const { sendSms } = require('../utils/sms');
+const { sendMail } = require('../utils/email');
+const models = require('../models');
+const Sequelize = require('sequelize');
+const Op = Sequelize.Op;
+const password = require('secure-random-password');
+// constants
+const SECURITY_CODE_DIGITS = 6;
+const twoFactorMethods = ['2fa-email', '2fa-phone'];
+const viewMethods = ['get_request'];
+const changeMethods = ['confirm'];
+const DETERM_KEY_SEED = process.env.DETERM_KEY_SEED || creatorKeyJson.private_key;
+const MULTISIG_CONTRACT_HASHES = process.env.MULTISIG_CONTRACT_HASHES ? process.env.MULTISIG_CONTRACT_HASHES.split() :['7GQStUCd8bmCK43bzD8PRh7sD2uyyeMJU5h8Rj3kXXJk','AEE3vt6S3pS2s7K6HXnZc46VyMyJcjygSMsaafFh67DF'];
+const CODE_EXPIRY = 300000;
+
+// generates a deterministic key based on the accountId
+const getKeyStore = (accountId) => ({
+    async getKey() {
+        const hash = crypto.createHash('sha256').update(accountId + DETERM_KEY_SEED).digest();
+        const keyPair = nacl.sign.keyPair.fromSeed(hash);
+        return nearAPI.KeyPair.fromString(base_encode(keyPair.secretKey));
+    },
+});
+
+// get the accountId's multisig contract instance
+const getContract = async (accountId) => {
+    const keyStore = getKeyStore(accountId);
+    const near = await nearAPI.connect({
+        deps: { keyStore },
+        nodeUrl: process.env.NODE_URL
+    });
+    const contractAccount = new nearAPI.Account(near.connection, accountId);
+    const contract = new nearAPI.Contract(contractAccount, accountId, {
+        viewMethods,
+        changeMethods,
+    }, '100000000000000');
```

Comment:
@mattlockyer nit: AFAIK `, '100000000000000'` is just ignored, there is no additional argument here:

https://github.com/near/near-api-js/blob/40d7f79c19174805e9c7500159f5166819305f5f/src/contract.ts#L22